src/plugins/data/public/search/aggs/metrics/serial_diff_fn.ts

```diff
@@ -0,0 +1,110 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { Assign } from '@kbn/utility-types';
+import { ExpressionFunctionDefinition } from '../../../../../expressions/public';
+import { AggExpressionType, AggExpressionFunctionArgs, METRIC_TYPES } from '../';
+import { getParsedValue } from '../utils/get_parsed_value';
+
+const fnName = 'aggSerialDiff';
+
+type Input = any;
+type AggArgs = AggExpressionFunctionArgs<typeof METRIC_TYPES.SERIAL_DIFF>;
+type Arguments = Assign<AggArgs, { metricAgg?: string; customMetric?: AggExpressionType }>;
+type Output = AggExpressionType;
+type FunctionDefinition = ExpressionFunctionDefinition<typeof fnName, Input, Arguments, Output>;
+
+export const aggSerialDiff = (): FunctionDefinition => ({
+  name: fnName,
+  help: i18n.translate('data.search.aggs.function.metrics.serial_diff.help', {
+    defaultMessage: 'Generates a serialized agg config for a serial_diff agg',
+  }),
+  type: 'agg_type',
+  args: {
+    id: {
+      types: ['string'],
+      help: i18n.translate('data.search.aggs.metrics.serial_diff.id.help', {
+        defaultMessage: 'ID for this aggregation',
+      }),
+    },
+    enabled: {
+      types: ['boolean'],
+      default: true,
+      help: i18n.translate('data.search.aggs.metrics.serial_diff.enabled.help', {
+        defaultMessage: 'Specifies whether this aggregation should be enabled',
+      }),
+    },
+    schema: {
+      types: ['string'],
+      help: i18n.translate('data.search.aggs.metrics.serial_diff.schema.help', {
+        defaultMessage: 'Schema to use for this aggregation',
+      }),
+    },
+    metricAgg: {
+      types: ['string'],
+      default: 'custom',
+      help: i18n.translate('data.search.aggs.metrics.serial_diff.metricAgg.help', {
+        defaultMessage:
+          'Id for finding agg config to use for building parent pipeline aggregations',
+      }),
+    },
+    customMetric: {
+      types: ['agg_type'],
+      help: i18n.translate('data.search.aggs.metrics.serial_diff.customMetric.help', {
+        defaultMessage: 'Agg config to use for building parent pipeline aggregations',
+      }),
+    },
+    buckets_path: {
+      types: ['string'],
+      required: true,
+      help: i18n.translate('data.search.aggs.metrics.serial_diff.buckets_path.help', {
+        defaultMessage: 'Path to the metric of interest',
+      }),
+    },
+    json: {
+      types: ['string'],
+      help: i18n.translate('data.search.aggs.metrics.serial_diff.json.help', {
+        defaultMessage: 'Advanced json to include when the agg is sent to Elasticsearch',
+      }),
+    },
+  },
+  fn: (input, args) => {
+    const { id, enabled, schema, ...rest } = args;
+
+    // Need to spread this object to work around TS bug:
+    // https://github.com/microsoft/TypeScript/issues/15300#issuecomment-436793742
+    const customMetric = args.customMetric?.value ? { ...args.customMetric.value } : undefined;
```

Comment:
I hope after merging #64772 we can remove it

Reply:
removed